File: SnipeMobile/API/HTMLDecoder.swift
    import Foundation

// Helper class for decoding HTML
class HTMLDecoder {
    static func decode(_ htmlString: String) -> String {
        // 1. Strip HTML tags
        var result = htmlString.replacingOccurrences(of: "<[^>]+>", with: "", options: .regularExpression, range: nil)
        // 2. Decode named HTML entities
        let entities: [String: String] = [
            "&quot;": "\"",
            "&apos;": "'",
            "&amp;": "&",
            "&lt;": "<",
            "&gt;": ">",
            "&euro;": "€",
            "&nbsp;": " ",
            // Voeg hier meer entities toe indien nodig
        ]
        for (entity, character) in entities {
            result = result.replacingOccurrences(of: entity, with: character)
        }
        // 3. Decode numeric entities (zoals &#039;)
        let pattern = "&#(\\d+);"
        let regex = try? NSRegularExpression(pattern: pattern)
        let nsrange = NSRange(result.startIndex..<result.endIndex, in: result)
        var offset = 0
        regex?.enumerateMatches(in: result, options: [], range: nsrange) { match, _, _ in
            guard let match = match, match.numberOfRanges == 2,
                  let range = Range(match.range(at: 0), in: result),
                  let numRange = Range(match.range(at: 1), in: result),
                  let code = Int(result[numRange]),
                  let scalar = UnicodeScalar(code) else { return }
            let replacement = String(scalar)
            let start = result.distance(from: result.startIndex, to: range.lowerBound) + offset
            let end = result.distance(from: result.startIndex, to: range.upperBound) + offset
            result.replaceSubrange(result.index(result.startIndex, offsetBy: start)..<result.index(result.startIndex, offsetBy: end), with: replacement)
            offset += replacement.count - (end - start)
        }
        return result
    }
} 

File: Android/app/src/main/java/com/callandt/snipemobile/util/HtmlDecoder.kt
package com.callandt.snipemobile.util

/**
 * Strip HTML tags and decode common HTML entities, matching the iOS [HTMLDecoder].
 */
object HtmlDecoder {
    private val TAG_PATTERN = Regex("<[^>]+>")
    private val NUMERIC_ENTITY_PATTERN = Regex("&#(\\d+);")

    private val NAMED_ENTITIES = mapOf(
        "&quot;" to "\"",
        "&apos;" to "'",
        "&amp;" to "&",
        "&lt;" to "<",
        "&gt;" to ">",
        "&euro;" to "€",
        "&nbsp;" to " ",
    )

    fun decode(htmlString: String): String {
        var result = htmlString.replace(TAG_PATTERN, "")
        for ((entity, character) in NAMED_ENTITIES) {
            result = result.replace(entity, character)
        }
        result = NUMERIC_ENTITY_PATTERN.replace(result) { match ->
            val code = match.groupValues[1].toIntOrNull()
            if (code != null) {
                code.toChar().toString()
            } else {
                match.value
            }
        }
        return result
    }
}
